java-frontend/src/test/java/org/sonar/java/resolve/targets/TypeAnnotationOnMembers.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.resolve.targets;
+
+import java.util.List;
+
+public class TypeAnnotationOnMembers<C> {
+
+  // Note: bellow annotations @TypeAnnotation("t") are not present in our semantic
+  // BytecodeFieldVisitor.visitTypeAnnotation does not support typePath != null
+  // BytecodeMethodVisitor.visitTypeAnnotation does not support the METHOD_TYPE_PARAMETER case
```

Comment:
I struggle to understand what this comment means.

Reply:
And from what I can understand from it, it seems to me that the support of type annotation you fixed was not complete ?

Reply:
I changed and moved the comment to unit tests and visitors code to ease the understanding by being close to the context.